R/epichains.R

```diff
@@ -0,0 +1,283 @@
+#' Print an [`epichains`] object
+#'
+#' @param x An [`epichains`] object.
+#' @param ... Other parameters passed to [print()].
+#' @return Invisibly returns an [`epichains`]. Called for side-effects.
+#' @export
+print.epichains <- function(x, ...) {
+  format(x, ...)
+}
+
+#' Format method for epichains class
+#'
+#' @param x epichains object
+#' @param ... further arguments passed to or from other methods
+#' @return Invisibly returns an [`epichains`]. Called for printing side-effects.
+#' @export
+format.epichains <- function(x, ...) {
+  # check that x is an epichains object
+  validate_epichains(x)
+
+  # summarise the information stored in x
+  chain_info <- summary(x)
+
+  if (attributes(x)$chain_type == "chains_tree") {
+    writeLines(
+      c(
+        sprintf("`epichains` object"),
+
+        "< tree head (from first known ancestor) >\n"
+        )
+      )
+
+    # print head of the simulation output
+    print(head(x[!is.na(x$ancestor), ]))
```

Comment:
That's a good point. I actually wonder if it won't be best to sort by `sim_id` then `ancestor`, so that it's easier to see the chains within each simulation. I have implemented that here 6972ed74ef71ea3ac466c714c89525d273ad3d16.